static/js/components/widgets/SelectField.tsx

```diff
@@ -1,36 +1,57 @@
-import React, { useCallback, ChangeEvent } from "react"
+import React, { useCallback, ChangeEvent, useState } from "react"
 import Select from "react-select"
-import AsyncSelect from "react-select/async"
 import { isNil } from "ramda"
+import { AsyncPaginate, LoadOptions } from "react-select-async-paginate"
+import AsyncSelect from "react-select/async"
+import { isFeatureEnabled } from "../../util/features"
 
 export interface Option {
   label: string
   value: string
 }
 
+export interface Additional {
+  callback?: (options: Option[]) => void
+}
```

Comment:
> Is that correct?

Yes, your observations are correct. Additionally, the `callback` also affects `RelationField` through the `useWebsiteSelectOptions` hook.

`react-select` can either update the options through a [callback or a promise](https://react-select.com/async#loading-asynchronously). I don't know why we have used both ways. The code in the [website hook](https://github.com/mitodl/ocw-studio/blob/8544fd21e5537df359527cc522a5d0a70657fb72/static/js/hooks/websites.ts#L112) makes a debounced fetch if we are using a callback. So I assume we use the callback way when we need debounced search. 


`AsyncPaginate` has its own internal debounce feature. I forgot to enable it before. But since it came up, I'll configure it now.

> whether we are actually using that data anywhere we plan to keep.

Nope. We only need `Additional` to pass around the `callback`. Once we stop using `AsyncSelect` we won't need it anymore. At that point, we might also want to update the `useWebsiteSelectOptions` hook.

That being said, there will always be a space for `Additional` in the signatures. For example, the type `LoadOptions<Option, Option, Additional>` would become `LoadOptions<Option, Option, undefined>`.

Reply:
> That being said, there will always be a space for Additional in the signatures. For example, the type LoadOptions<Option, Option, Additional> would become LoadOptions<Option, Option, undefined>.

In the the `AsyncPaginate` prop `loadOptions`, yes. But I think `SelectFieldProps["loadOptions"]` could drop the 3rd arg once it's unused. Would make it clearer that we don't use the data. (Once we aren't using it, anyway).